server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java

```diff
@@ -0,0 +1,186 @@
+   /**
+    * Indicates we've started sending bytes.
+    */
+   private boolean started; // want to eliminate this
+
+   // TODO The use of this is very questionable!  I'm just throwing
+   // this in at the very end before I try something radically different
+   // for the entire adapter.
+   private Duration timeout;
+
+   /**
+    * The {@link Flux<byte[]>} that is fed into {@link HttpServerResponse#sendByteArray}.
+    */
+   private Flux<Tuple2<byte[], CompletableFuture<Boolean>>> out = Flux.create(sink -> {
+      log.trace("Establishing sink and listener!");
+      listener.set(new ChunkOutputStream.EventListener() {
+         @Override
+         public void data(final byte[] bs, final CompletableFuture<Boolean> demandedSignal) {
+             if (log.isTraceEnabled()) {
+                 log.trace("Sending {} bytes", bs.length);
+             }
+             sink.next(Tuples.of(bs, demandedSignal));
+         }
+         @Override
+         public void finish() {
+             sink.complete();
+         }
+      });
+      sinkCreated.complete(COMPLETED_SIGNAL);
+   });
+
+   /**
+    * The interface used to transfer data from an eventing API (i.e. {@link
+    * #asyncWrite}) to a {@link Flux} ({@link #out} in this case).  See
+    * documentation around {@link Flux#create} for more information on this.
+    */
+   interface EventListener {
+      void data(byte[] s, CompletableFuture<Boolean> demandedSignal);
+      void finish();
+   }
+
+   ChunkOutputStream(
+       final HttpServerResponse response,
+       final MonoProcessor<Void> completionMono
+   ) {
+      this.response = response;
+      this.completionMono = completionMono;
+   }
+
+   @Override
+   public void write(int b) {
+      listener.get().data(new byte[] {(byte)b}, new CompletableFuture<>());
+   }
+
+   @Override
+   public void close() throws IOException {
+       log.trace("Closing the ChunkOutputStream.");
+       final EventListener el = listener.get();
+       if (el != null) {
+           el.finish();
+       } else {
+           Mono.<Void>empty().subscribe(completionMono);
+       }
+       super.close();
+   }
+
+   @Override
+   public void write(byte[] bs, int off, int len) {
+       try {
+           asyncWrite(bs, off, len).get();
+       } catch (final InterruptedException ie) {
+           Thread.currentThread().interrupt();
+           throw new RuntimeException(ie);
+       } catch (final ExecutionException ee) {
+           throw new RuntimeException(ee);
+       }
+   }
+
+   @Override
+   public void flush() throws IOException {
+       log.trace("Flush called on ChunkOutputStream");
+      super.flush();
+   }
+
+   @Override
+   public CompletionStage<Void> asyncFlush() {
+      // TODO
+      return CompletableFuture.completedFuture(null);
+   }
+
+
+    @Override
+   public CompletableFuture<Void> asyncWrite(final byte[] bs, int offset, int length) {
+        if (!started) {
+            started = true;
+            Flux<byte[]> actualOut =
```

Comment:
So, this is a `Flux` because we expect `asyncWrite` to be called multiple time for a response?  I assume that's true for SSE scenarios, or potentially other streaming/chunking scenarios.

Reply:
I haven't tested SSE, but yes it happens for chunking scenarios.